activesupport/lib/active_support/isolated_execution_state.rb

```diff
@@ -1,14 +1,11 @@
 # frozen_string_literal: true
 
 module ActiveSupport
-  module IsolatedExecutionState # :nodoc:
-    @isolation_level = nil
-
-    Thread.attr_accessor :active_support_execution_state
-    Fiber.attr_accessor :active_support_execution_state
+  class IsolatedExecutionState # :nodoc:
```

Comment:
Non-blocking, but would it make sense to make this the instance, ie...

```
class IsolatedExecutionStateContainer # not sure about naming 😅
   ...
end

IsolatedExecutionState = IsolatedExecutionStateContainer.new
```

That way we'd avoid the extra method calls from delegation on this heavily used path